trinity/nodes/base.py

```diff
@@ -9,39 +12,71 @@
 from p2p.service import (
     BaseService,
     EmptyService,
+    ServiceContext,
+)
+from trinity.chains import (
+    ChainProxy,
+)
+from trinity.chains.header import (
+    AsyncHeaderChainProxy,
 )
+from trinity.db.chain import ChainDBProxy
+from trinity.db.base import DBProxy
+from trinity.db.header import AsyncHeaderDBProxy
 from trinity.rpc.main import (
     RPCServer,
 )
 from trinity.rpc.ipc import (
     IPCServer,
 )
+from trinity.config import (
+    ChainConfig,
+)
 
 
 class Node(BaseService):
     """
     Create usable nodes by adding subclasses that define the following
-    unset attributes...
+    unset attributes.
     """
-
     chain_class: Type[BaseChain] = None
-    peer_chain_class: Type[BaseService] = None
 
-    def __init__(
-            self,
-            headerdb: BaseHeaderDB,
-            peer_pool,
-            jsonrpc_ipc_path: Path = None):
+    def __init__(self, chain_config: ChainConfig) -> None:
         super().__init__()
-        self._headerdb = headerdb
-        self._jsonrpc_ipc_path = jsonrpc_ipc_path
-        self._peer_pool = peer_pool
-        self._peer_chain = self.peer_chain_class(headerdb, peer_pool)
+
+        self._db_manager = create_db_manager(chain_config.database_ipc_path)
+        self._db_manager.connect()  # type: ignore
+        self._headerdb = self._db_manager.get_headerdb()  # type: ignore
+
+        self._jsonrpc_ipc_path: Path = chain_config.jsonrpc_ipc_path
+        self._auxiliary_services = ServiceContext()
 
     @abstractmethod
     def get_chain(self) -> BaseChain:
         raise NotImplementedError("Node classes must implement this method")
 
+    @abstractmethod
+    def get_p2p_server(self) -> BaseService:
+        """
+        This is the main service that will be run, when calling :meth:`run`.
+        It's typically responsible for syncing the chain, with peer connections.
```

Comment:
Thoughts on eventually combining `p2p.Server` with `trinity.Node`?

Reply:
I think it makes sense for there to be an orchestration service in p2p (like `Server`) that manages the several services it needs to run. But I also think it makes sense to have an umbrella orchestration service in trinity (like `Node`), which manages several services (including `Server`). My instinct is that some services should not be managed inside p2p, but I think we already have one example: the RPC server.

But... there are some things handled in `Node` right now that could/should be pushed into `Server` (especially the light node, which explicitly manages the peer pool, at the moment).